content/guides/understanding-los-records.md

```diff
@@ -0,0 +1,129 @@
+# Understanding LOS Records
+
+## What is LOS?
+
+LOS stands for Length Of Stay and LOS Records are a simple but powerful way to define rates.
+
+It's best explained through examples.
+
+## (1) Basic example
+Let's take an LOS Record:
+
+~~~
+day: 2000-01-01
+min_occupancy: 1
+max_occupancy: 4
+rates: 50,100,150,200,250,300,350,400,450,500,...
+~~~
+
+### Meaning
+
+When the check-in date is 2000-01-01 and the number of guests is between 1 and 4 (inclusive), then the price is:
+
+~~~
+50 for 1 night
+100 for 2 nights
+150 for 3 nights
+...
+500 for 10 nights
+~~~
+
+---
+
+## (2) Additional guest
+
+~~~
+day: 2000-01-01
+min_occupancy: 1
+max_occupancy: 4
+rates: 50,100,150,200,250,300,350,400,450,500,...
+
+day: 2000-01-01
+min_occupancy: 5
+max_occupancy: 5
+rates: 60,120,180,240,300,360,420,480,540,600,...
+~~~
+
+### Meaning
+
+We can see that the daily price increased by 10 from adding an extra guest.
+
+---
+
+## (3) No check-in
+
+~~~
+day: 2000-01-01
+rates: 0,0,0,0,0,...,0
+~~~
+
+### Meaning
+
+You can't check in on 2000 January 1.
+
+---
+
+## (4) Minimum stay
+
+~~~
+day: 2000-01-01
+rates: 0,0,150,200,250,300,350,400,450,500,...
+~~~
+
+### Meaning
+
+When checking in on 2000 January 1 you must stay for at least 3 nights.
+
+---
+
+## (5) Maximum stay
+
+~~~
+day: 2000-01-01
+rates: 50,100,150,200,250,300,350,0,...,0
+~~~
+
+### Meaning
+
+When checking in on 2000 January 1 you can't stay for more than 7 nights.
```

Comment:
This needs to be double checked, what is there is check-out only rule that allows to checkout only e.g. on weekends? In that case, I think it's possible to have some 0s in-between on such case.

Reply:
Huh? I'm just explaining the basics here, don't complicate it. :D

Reply:
Ok, but I think this might simple be not true.

Reply:
What's not true? When you see: `100,200,0,...0` then it means that maximum stay is 2 nights.

Reply:
What if there will be:
```
100, 200, 0, 0, 0, 0, 0, 0, 0, 1000
```
? 

Isn't it possible with some specific combination of rules, especially checkout only?

Reply:
I think for Karol it's not explicit (and I have to agree here) that `0,...,0` means there are only zeros within this range.

Reply:
That's what it means, only zeros. I recall they accepted this kind of notation at university math. It's also the most obvious interpretation.

Reply:
The notation is fine for clearly defined series, but here, I though it's like `put whatever here`, not not about 0s. And this is quite important as this might lead to wrong conclusions how can maximum stay be resolved from LOS records.

Reply:
:doge: so should I put 30 zeroes there or what?

Reply:
I would just mention the case where you need to be careful with the interpretation. I actually oversimplified it in HA-PM when I was resolving maximum stay and didn't consider that rule :doge:

Reply:
This is just about understanding how LOS works. The example states that if there are only zero values for rates after a X nights, then it could be translated as the maximum stay for that day is X.